src/shared/src/types/bom-types.ts

```diff
@@ -0,0 +1,82 @@
+import { User, UserPreview } from './user-types';
+
+export enum MaterialStatus {
+  Ordered = 'ORDERED',
+  Received = 'RECEIVED',
+  Unordered = 'UNORDERED',
+  Shipped = 'SHIPPED'
+}
+
+export interface Unit {
+  name: string;
+  materials: MaterialPreview;
+}
+
+export type UnitPreview = Omit<Unit, 'materials'>;
+
+export interface MaterialType {
+  name: string;
+  dateCreated: Date;
+  creatorId: number;
+  creator: UserPreview;
+  materials: MaterialPreview[];
+}
+export type MaterialTypePreview = Omit<MaterialType, 'materials' | 'creator'>;
+
+export interface Assembly {
+  assemblyId: string;
+  name: string;
+  pdmFileName?: string;
+  dateDeleted?: Date;
+  userDeletedId?: number;
+  userDeleted?: UserPreview;
+  userCreatedId: number;
+  userCreated: UserPreview;
+  wbsElementId: number;
+  materials: MaterialPreview[];
+}
+
+export type AssemblyPreview = Omit<Assembly, 'materials'>;
+
+export interface Manufacturer {
+  name: string;
+  dateCreated: Date;
+  creatorId: number;
+  creator: User;
+  materials: MaterialPreview[];
```

Comment:
a good chunk of these types exist in projects-types rn fyi, we'll need to make sure every place that was using those uses these now, and that the shared types actually make sense

Reply:
Yesh i deleted them